core/src/main/scala/cats/Applicative.scala

```diff
@@ -67,8 +68,84 @@ import simulacrum.typeclass
 object Applicative {
   def monoid[F[_], A](implicit f: Applicative[F], monoid: Monoid[A]): Monoid[F[A]] =
     new ApplicativeMonoid[F, A](f, monoid)
+
+  /**
+   * Creates an applicative functor for `F`, holding domain fixed and combining
+   * over the codomain.
+   *
+   * Example:
+   * {{{
+   * scala> import cats.implicits._
+   * scala> import cats.Applicative.catsApplicativeForArrow
+   * scala> val toLong: Int => Long = _.toLong
+   * scala> val double: Int => Int = 2*_
+   * scala> val f: Int => (Long, Int) = catsApplicativeForArrow.product(toLong, double)
+   * scala> f(3)
+   * res0: (Long, Int) = (3,6)
+   * }}}
+   */
+  implicit def catsApplicativeForArrow[F[_, _], A](implicit F: Arrow[F]): Applicative[F[A, ?]] =
+    new ArrowApplicative[F, A](F)
 }
 
 private[cats] class ApplicativeMonoid[F[_], A](f: Applicative[F], monoid: Monoid[A]) extends ApplySemigroup(f, monoid) with Monoid[F[A]] {
   def empty: F[A] = f.pure(monoid.empty)
 }
+
+private[cats] class ArrowApplicative[F[_, _], A](F: Arrow[F]) extends Applicative[F[A, ?]] {
+  /**
+   * Pure for Arrows lifts the constant function at that value.
+   * Example:
+   * {{{
+   * scala> import cats.implicits._
+   * scala> import cats.Applicative.catsApplicativeForArrow
+   * scala> val g: Int => String = catsApplicativeForArrow.pure("hello")
+   * scala> g(5)
+   * res0: String = hello
+   * }}}
+   */
```

Comment:
Thanks for writing these doctests. But doctests in an instance are probably not necessary. They won't be included in apidoc, and these methods are already thoroughly tested in law tests of the instance. So, although the more tests the better, having them here has two slight cons, a) more code to read, in an instance definition, people want to quickly see how the instance is the implemented, rather than the API usage. and b) they provide redundant test coverage for these implementations, so if somehow the instance is not test by law tests, the test coverage report won't expose that.

Reply:
Ah, yeah, that makes sense.